src/utils/diff.ts

```diff
@@ -0,0 +1,54 @@
+import { set, isEmpty, isEqual, isNil as isNullOrUndefined } from 'lodash';
+import { Dictionary } from '../configuration/interfaces';
+
+const OBJECTS_EMPTY_ERROR                = 'objects should be not empty';
```

Comment:
Error messages should start from an uppercase letter.

Reply:
These errors are not TestCafe errors. 
We need them to immediately stop reccursive function on the deepest level.
I think, that this options comparison should never throw error in TestCafe, so we do not need `GeneralError` as well.
If the comparison is failed for some reason I just plan to catch the error and return null. However, in the current implementation compared objects are always of the same type and not empty. I added this validation to save us from future changes.